packages/next/src/server/app-render/app-render.tsx

```diff
@@ -615,15 +615,69 @@ async function renderToHTMLOrFlightImpl(
     enableTainting,
   } = renderOpts
 
+  // Combined load times for loading client components
+  let clientComponentLoadStart = 0
+  let clientComponentLoadTimes = 0
+  let clientComponentLoadCount = 0
+
   // We need to expose the bundled `require` API globally for
   // react-server-dom-webpack. This is a hack until we find a better way.
   if (ComponentMod.__next_app__) {
     // @ts-ignore
-    globalThis.__next_require__ = ComponentMod.__next_app__.require
+    globalThis.__next_require__ = (...args: any[]) => {
+      if (clientComponentLoadStart === 0) {
+        clientComponentLoadStart = Date.now()
+      }
+
+      const startTime = Date.now()
```

Comment:
One assumption that we are making here is that all of these require times are sequential. Is that true? Otherwise we will get bigger values than that it actually takes.